snuba/datasets/errors_replacer.py

```diff
@@ -526,8 +541,42 @@ def process_message(self, message: ReplacementMessage) -> Optional[Replacement]:
 
         return processed
 
+    def _message_already_processed(self, message: ReplacementMessage) -> bool:
```

Comment:
WHat is the reasoning for putting this login in `errors_replacer.py` in stead of the more abstract `replacer.py`?
https://github.com/getsentry/snuba/blob/master/snuba/replacer.py#L334-L383

I think there is a case for moving it in the `ReplacerWorker`. 
- errors_replacer is independent from Kafka, the replacement object does not need to know about offsets or partitions. So if you move it to replacer.py you would not have to introduce these concerns here. replacer.py instead knows about kafka
- if you move it to replacer.py you will be able to reuse this logic in other replacements (beyond errors).

Reply:
I had this logic written in `ReplacerWorker` to start with actually, but at the worker level I do not have access to the Replacer state (events/errors). I moved the logic into `errors_replacer.py` to get the state to use as part of the Redis key.

I think I can create method like `get_state()` on the base `ReplacerProcessor` class to get around this, if that would be a better way to do it.

Reply:
Yes, I think exposing the state to the parent class will be better than exposing Kafka details to the worker and make it errors specific.